tests/integration/replace.go

```diff
@@ -0,0 +1,60 @@
+package tests
+
+import (
+	"bytes"
+	"html/template"
+	"maps"
+	"strconv"
+	"strings"
+
+	"github.com/sourcenetwork/defradb/errors"
+	"github.com/sourcenetwork/defradb/tests/state"
+	"github.com/stretchr/testify/require"
+)
+
+// templateDataGenerators contains a set of data generators by their template prefix.
+//
+// Supporting action properties will replace any templated elements with data drawn from these
+// sets.
+var templateDataGenerators = map[string]func(*state.State, int) map[string]string{
+	"CID": func(s *state.State, nodeID int) map[string]string {
+
+		docIDsToCIDs := s.Nodes[nodeID].Composites
+
+		res := map[string]string{}
+		for colIndex, docIndexes := range s.DocIDs {
+			for dIndex, docID := range docIndexes {
+				cids := docIDsToCIDs[docID.String()]
+				for cidIndex, cid := range cids {
+					res["CID"+strconv.Itoa(colIndex)+"_"+strconv.Itoa(dIndex)+"_"+strconv.Itoa(cidIndex)] = cid.String()
```

Comment:
nit:

would the code be clearer here if this long string concatenation were assigned to a local variable, named for example `templateCIDRef` or something, and then used as `res[templateCIDRef] = cid.String()`?